includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -210,21 +210,33 @@ public function get_button_height() {
 	 *
 	 * @param object $product WC_Product_* object.
 	 * @return mixed Total price.
+	 *
+	 * @throws Invalid_Price_Exception Whenever a product has no price.
 	 */
 	public function get_product_price( $product ) {
-		$product_price = $product->get_price();
-
 		// If prices should include tax, using tax inclusive price.
 		if ( ! $this->prices_exclude_tax() ) {
-			$product_price = wc_get_price_including_tax( $product );
+			$base_price = wc_get_price_including_tax( $product );
+		} else {
+			$base_price = $product->get_price();
 		}
 
 		// Add subscription sign-up fees to product price.
-		if ( 'subscription' === $product->get_type() && class_exists( 'WC_Subscriptions_Product' ) ) {
-			$product_price = $product_price + WC_Subscriptions_Product::get_sign_up_fee( $product );
+		$sign_up_fee        = 0;
+		$subscription_types = [
+			'subscription',
+			'subscription_variation',
+		];
+		if ( in_array( $product->get_type(), $subscription_types, true ) && class_exists( 'WC_Subscriptions_Product' ) ) {
+			// When there is no sign-up fee, `get_sign_up_fee` falls back to an int 0.
+			$sign_up_fee = WC_Subscriptions_Product::get_sign_up_fee( $product );
 		}
 
-		return $product_price;
+		if ( ! is_numeric( $base_price ) || ! is_numeric( $sign_up_fee ) ) {
```

Comment:
Do we need a check for the numeric `$sign_up_fee` here? We set the initial value to `0` and it also falls back to an int `0` so will there be a case where this is non numeric?

Reply:
My thinking here was to check both, just to be on the safe side. I saw that upon an empty string for the signup fee, `WC_Subscriptions_Product::get_sign_up_fee()` would happen to cast the empty string to false, and return the default value, which is `int(0)`, but that felt more like a lucky and unreliable side effect, rather than intentional casting.